module/users/helper.js

```diff
@@ -321,149 +327,123 @@ module.exports = class UsersHelper {
            *
            * @return [{Objects}] array of solutions documents
            */
-          mergedData = await solutionsHelper.solutionDocuments(
-            { _id: { $in: solutionIds } },
-            [
-              "name",
-              "description",
-              "programName",
-              "programId",
-              "externalId",
-              "projectTemplateId",
-              "type",
-              "language",
-              "creator",
-              "endDate",
-              "link",
-              "referenceFrom",
-              "entityType",
-              "certificateTemplateId",
-            ]
-          );
-          totalCount = mergedData.length;
-        } else {
-          let autoTargetedSolutions =
-            await solutionsHelper.forUserRoleAndLocation(
-              requestedData,
-              type,
-              "",
-              programId,
-              constants.common.DEFAULT_PAGE_SIZE,
-              constants.common.DEFAULT_PAGE_NO,
-              search
+          const solutionsWithPreviousProfile =
```

Comment:
@ankitpws Please include comments for better understanding of logic

Reply:
done